pkg/controller/controller_binding.go

```diff
@@ -594,69 +608,45 @@ func setServiceInstanceCredentialConditionInternal(toUpdate *v1alpha1.ServiceIns
 	toUpdate.Status.Conditions = append(toUpdate.Status.Conditions, newCondition)
 }
 
-func (c *controller) updateServiceInstanceCredentialStatus(toUpdate *v1alpha1.ServiceInstanceCredential) error {
-	glog.V(4).Infof("Updating status for ServiceInstanceCredential %v/%v", toUpdate.Namespace, toUpdate.Name)
-	_, err := c.serviceCatalogClient.ServiceInstanceCredentials(toUpdate.Namespace).UpdateStatus(toUpdate)
-	if err != nil {
-		glog.Errorf("Error updating status for ServiceInstanceCredential %v/%v", toUpdate.Namespace, toUpdate.Name)
+// setServiceInstanceCredentialForPendingChange sets the status of the
+// ServiceInstanceCredential to indicate that there is a pending change that is
+// being processed in a retry loop
+func (c *controller) setServiceInstanceCredentialForPendingChange(toUpdate *v1alpha1.ServiceInstanceCredential) {
+	glog.V(5).Infof("Setting ServiceInstanceCredential '%v/%v' for pending change", toUpdate.Namespace, toUpdate.Name)
+	if toUpdate.Status.OperationStartTime == nil {
+		now := metav1.Now()
+		toUpdate.Status.OperationStartTime = &now
 	}
-	return err
 }
 
-// updateServiceInstanceCredentialCondition updates the given condition for the given ServiceInstanceCredential
-// with the given status, reason, and message.
-func (c *controller) updateServiceInstanceCredentialCondition(
-	binding *v1alpha1.ServiceInstanceCredential,
-	conditionType v1alpha1.ServiceInstanceCredentialConditionType,
-	status v1alpha1.ConditionStatus,
-	reason, message string) error {
-
-	toUpdate, err := makeServiceInstanceCredentialClone(binding)
-	if err != nil {
-		return err
-	}
-
-	c.setServiceInstanceCredentialCondition(toUpdate, conditionType, status, reason, message)
+// setServiceInstanceCredentialForCompletedChange sets the status of the
+// ServiceInstanceCredential to indicate that a pending change has been completed
+func (c *controller) setServiceInstanceCredentialForCompletedChange(toUpdate *v1alpha1.ServiceInstanceCredential) {
```

Comment:
"setThingForCompleted" doesn't read quite right to me - how about 'clearOperationFor' or something like that?

Reply:
@staebler I agree. I sort of got used to the names out of habit, but I think they could be more clear. My suggestion is:

`setServiceInstanceStatusStartingOperation` and `setServiceInstanceStatusFinishedOperation` or something similar.

Reply:
My issue with using "clearOperationFor" is that the function is doing more than just clearing. It is also setting the ReconciledGeneration. In the end, though, I don't particularly care too much what it is called.